tablo/models.py

```diff
@@ -1032,94 +991,85 @@ def __init__(self, **entries):
 
 
 def create_aggregate_database_table(row, dataset_id):
-    row.append(Column(column=SOURCE_DATASET_FIELD_NAME, type='string'))
-    optional_fields = [col.column for col in row if hasattr(col, 'required') and not col.required]
-    table_name = create_database_table(row, dataset_id, optional_fields=optional_fields)
-    row.pop()  # remove the appended column
-    return table_name
-
-
-def create_database_table(row, dataset_id, append=False, optional_fields=None):
-    optional_fields = optional_fields or []
-    table_name = TABLE_NAME_PREFIX + dataset_id + IMPORT_SUFFIX
-
-    if not append:
-        drop_table_command = 'DROP TABLE IF EXISTS {table_name}'.format(table_name=table_name)
-        create_table_command = 'CREATE TABLE {table_name} ({pk} serial NOT NULL PRIMARY KEY'.format(
-            table_name=table_name,
-            pk=PRIMARY_KEY_NAME
+    columns = [convert_header_to_column_name(SOURCE_DATASET_FIELD_NAME)]
+    data_types = ['String']
+    optional_fields = []
+    type_conversion = {
+        'decimal': 'Decimal',
+        'date': 'Date',
+        'integer': 'Integer',
+        'string': 'String',
+        'xLocation': 'Decimal',
+        'yLocation': 'Decimal',
+        'dropdownedit': 'String',
+        'dropdown': 'String',
+    }
+    for col in row:
+        col_name = convert_header_to_column_name(col.column)
+        columns.append(col_name)
+        data_types.append(type_conversion[col.type])
+        if hasattr(col, 'required') and not col.required:
+            optional_fields.append(col_name)
+    df = pd.DataFrame(columns=columns)
+    df_info = {
+        'dataTypes': data_types,
+        'optionalFields': optional_fields
+    }
+    table_name = create_database_table(df, df_info, dataset_id)
+    with get_sqlalchemy_engine().connect() as conn:
+        sequence_name = '{}_0_seq'.format(table_name)
+        conn.execute('CREATE SEQUENCE {}'.format(sequence_name))
+        conn.execute(
+            'ALTER TABLE {table} ALTER COLUMN {key} TYPE bigint USING {key}::bigint'.format(
+                table=table_name, key=PRIMARY_KEY_NAME
+            )
         )
-        type_conversion = {
-            'decimal': 'double precision',
-            'date': 'date',
-            'integer': 'bigint',
-            'string': 'text',
-            'xlocation': 'double precision',
-            'ylocation': 'double precision',
-            'dropdownedit': 'text',
-            'dropdown': 'text',
-            'double': 'double precision',
-            'image': 'text'
-        }
-        for cell in row:
-            create_table_command += ', {field_name} {type} {null}'.format(
-                field_name=cell.column,
-                type=type_conversion[re.sub(r'\([^)]*\)', '', str(cell.type).lower())],
-                null='NOT NULL' if cell.column not in optional_fields else ''
+        conn.execute(
+            'ALTER TABLE {} ALTER COLUMN {} SET DEFAULT nextval(\'{}\')'.format(
```

Comment:
I suggest using named format keys here, as you do above.

Reply:
Done